mango/clients/aemet.py

```diff
@@ -0,0 +1,598 @@
+        if self._wait_time < self._DEFAULT_WAIT_TIME:
+            logging.warning(
+                "Wait time is too low. API limits may be exceeded. Minimum wait time is {} seconds.".format(
+                    self._DEFAULT_WAIT_TIME
+                )
+            )
+        logging.info(
+            "Aemet API has a limit of 50 requests per minute. Please be patient. Wait time is set to {} seconds.".format(
+                self._wait_time
+            )
+        )
+
+    def connect(self) -> None:
+        """
+        This method will connect to the AEMET API and cache the stations and municipios.
+        Checks if the API key is valid.
+        :doc-author: baobab soluciones
+        """
+        # Allows to check if the api_key is valid and cache the stations
+        try:
+            self._all_stations = self._get_all_stations()
+        except ApiKeyError as e:
+            raise e
+        # Setup municipios
+        self.municipios = self._get_municipios()
+
+    @property
+    def all_stations(self) -> List[dict]:
+        """
+        This property will return all the meteorological stations in Spain.
+        :return: List of dictionaries with the meteorological stations
+        :doc-author: baobab soluciones
+        """
+        return self._all_stations
+
+    @property
+    def municipios(self) -> List[dict]:
+        """
+        This property will return all the municipios in Spain.
+        :return: List of dictionaries with the municipios
+        :doc-author: baobab soluciones
+        """
+        return self._municipios
+
+    @municipios.setter
+    def municipios(self, value) -> None:
+        """
+        Keep only the relevant information for the municipios
+        """
+        tmp = []
+        for m in value:
+            m["codigo postal"] = m["id"].lstrip("id")
+            tmp.append(
+                {
+                    k: v
+                    for k, v in m.items()
+                    if k in ["codigo postal", "nombre", "longitud_dec", "latitud_dec"]
+                }
+            )
+        self._municipios = tmp
+
+    @staticmethod
+    def _haversine(point1, point2) -> float:
+        """
+        Calculate the great circle distance in kilometers between two points
+        on the earth (specified in decimal degrees).
+        :param point1: Tuple with latitude and longitude of the first point
+        :param point2: Tuple with latitude and longitude of the second point
+        """
+        # convert decimal degrees to radians
+        lon1, lat1 = point1
+        lon2, lat2 = point2
+        lon1, lat1, lon2, lat2 = map(radians, [lon1, lat1, lon2, lat2])
+
+        # haversine formula
+        dlon = lon2 - lon1
+        dlat = lat2 - lat1
+        a = sin(dlat / 2) ** 2 + cos(lat1) * cos(lat2) * sin(dlon / 2) ** 2
+        c = 2 * asin(sqrt(a))
+        r = 6371  # Radius of earth in kilometers. Use 3956 for miles. Determines return value units.
+        return c * r
```

Comment:
Why not use the implementation of geopy for the distance?

Reply:
To reduce dependencies. The error using this aproximation is low enough for the needs (I have tested it "enough")  If you prefer we can add geopy to dependencies